app/models/concerns/catalyst/input_parameterizable.rb

```diff
@@ -0,0 +1,50 @@
+module Catalyst
+  module InputParameterizable
+    extend ActiveSupport::Concern
+
+    included do
+      # JSON serialization for SQLite compatibility
+      serialize :input_params, coder: JSON
+    end
+
+    # Helper methods for input parameter management
+    def input_parameters
+      input_params || {}
+    end
+
+    def input_parameters=(params)
+      self.input_params = params&.transform_keys(&:to_s)
+    end
+
+    # Helper method to get specific input parameter
```

Comment:
I am not big fan of these helper methods, maybe we don't need a concern at all. Input params is just to store the input parameter values as json in the table. Remove the following helpers.

Additionally, consider removing this concern, if it would be composed only of a single `serialize` call.

Additionally, repeat above considerations and relevant comments for ModelParameterizable concern 

:point_up: each refactor in separate commit (TDD approach)